### DISCUSSION TITLE
useFallbackUrl doesn't work

### DISCUSSION THREAD

**Discussion by homeoftheunits** [spatie/laravel-medialibrary#2734] [Q&A]
Hi Guys,

i use medialibary 9.11.1 with Laravel 8. Runs in an sail enviroment with php8 and minIO.

In my Model i register a Media Collection:
```
public function registerMediaCollections(): void
{
    $this
        ->addMediaCollection('logo')
        ->useDisk('s3')
        ->useFallbackUrl($this->getFallbackLogoUrl())
        ->acceptsMimeTypes($this->acceptedImageMimeTypes)
        ->singleFile();
}

protected function getFallbackLogoUrl(): string
{
    if ($this->name !== null) {
        return
            sprintf(
                '%s%s%s%s',
                'https://ui-avatars.com/api/?name=',
                urlencode($this->name),
                '&color=ffffff&background=',
                $this->icon_color
            );
    }

    return '';
}
```

When the model has an media, everything is fine. When the model has no media, it should return the value from getFallbackLogoUrl(), but it throws the error:
```
PHP Error:  Call to a member function getUrl() on null in /var/www/html/app/Models/Application.php on line 97
```

Cheers
Ralf

**Comment by freekmurze** [comment]
Could your PR and/or a fix for this?

**Comment by homeoftheunits** [comment]
Sorry, i have no fix for this

**Comment by danielsimkus** [comment]
Can you provide more context around the error:
```
PHP Error:  Call to a member function getUrl() on null in /var/www/html/app/Models/Application.php on line 97
```
Are you using `getFirstMediaUrl` or trying to load the first instance of the MediaCollection and then using `getUrl()` as not sure if that works 🤷

**Reply by homeoftheunits** [reply]
Hi,

i use it in the model like this:
```
return $this
     ->getFirstMedia('logo')
     ->getUrl('thumb');
```

Why should this not work? In my understanding `useFallbackUrl` should return a Fallback, if no image for this model exist.

**Reply by danielsimkus** [reply]
I can't say for certain, I've never used that feature myself.
But judging by the docs, it states the following:
![Screenshot 2022-01-19 at 09 41 15](https://user-images.githubusercontent.com/8949315/150104795-4c5a51e8-dab2-45db-b28c-152e9c4a14a1.png)
which makes me think it only works when using the `getFirstMediaUrl` method maybe? 🤷

**Comment by freekmurze** [comment]
I'm moving this to a discussion as we're not actively going to work on this. Feel free to PR a failing test (and/or fix) for this problem.

**Comment by theahmedy** [comment]
I don't know your senario but I think using API for just display avatar filler it could be effeciantly with front end by using css

**seudo code**

```
@if($model->getFirstMediaUrl('logo')  == '' )
<img src="{{get image url from $model->getFirstMediaUrl('logo') }}" alt="...">
@else
<div class="avatar-initials"> get first & second charachter of  $user name</div>
@endif
```
**somewhere in your css**
```
.avatar-initials{
    background-color: #4273fa;
    color: #fff;
    display: inline-block;
    margin: 0 10px;
    letter-spacing: 0;
    width: 35px;
    height: 35px;
    line-height: 35px;
    border-radius: 100%;
    text-align: center;
}
```
